go.mod

```diff
@@ -25,7 +25,7 @@ require (
 	github.com/zclconf/go-cty v1.14.4
 	go.mongodb.org/atlas v0.36.0
 	go.mongodb.org/atlas-sdk/v20231115014 v20231115014.0.0
-	go.mongodb.org/atlas-sdk/v20240530001 v20240530001.0.1-0.20240628150242-4d7b444ed046
+	go.mongodb.org/atlas-sdk/v20240530002 v20240530002.0.1-0.20240703083409-9e77f6cad45f
```

Comment:
- `v20231115014` is preserved to old cluster endpoints
- `v20240530002` is adjusted to use lastest preview which includes new cluster API